### PR TITLE
#4212 Use plural form with caches_remove_all_confirm 

### PR DESCRIPTION
**Pull request by madankumarbaskaran**
Please find the pull request for issue #4212. strings.xml and CacheListActivity.java file is changed for this pull request.

### CODE REVIEW

**Comment by Lineflyer** [comment]
It sounds logical to me that the string for "one" should not be "all 1 caches" but e.g. "this cache".

File: main/res/values/strings.xml [hunk 1]
```diff
@@ -297,7 +297,10 @@
     <string name="caches_nearby">Nearby</string>
     <string name="caches_manage">Manage</string>
     <string name="caches_remove_all">Remove all</string>
-    <string name="caches_remove_all_confirm">Do you want to remove all %s caches from current list?</string>
+    <plurals name="caches_remove_all_confirm">
+        <item quantity="one">Do you want to remove all %s cache from current list?</item>
```
Comment (Lineflyer) [inline main/res/values/strings.xml]: It sounds logical to me that the string for "one" should not be "all 1 cache" but e.g. "this cache".

**Comment by samueltardieu** [comment]
`caches_remove_selected_confirm` should be treated the same way.

**Comment by madankumarbaskaran** [comment]
Hi Lineflyer/samueltardie,
                                       I will change as per your suggestion. Please let me know for any other change in my fix.

Thanks
Madan

**Comment by madankumarbaskaran** [comment]
Hi Lineflyer/samueltardie,
   Plurals are added for cache_remove_all_confirm and cache_remove_selected_confirm. Please review my pull request.

Thanks
Madan

File: main/src/cgeo/geocaching/CacheListActivity.java [hunk 2]
```diff
@@ -1166,8 +1166,9 @@ public void onClick(final DialogInterface dialog, final int which) {
 
     private void dropStored() {
         final int titleId = (adapter.getCheckedCount() > 0) ? R.string.caches_remove_selected : R.string.caches_remove_all;
-        final int messageId = (adapter.getCheckedCount() > 0) ? R.string.caches_remove_selected_confirm : R.string.caches_remove_all_confirm;
-        final String message = getString(messageId, adapter.getCheckedOrAllCount());
+        final int count= adapter.getCheckedOrAllCount();
+        final String message = (adapter.getCheckedCount() > 0) ?
+                res.getQuantityString(R.plurals.caches_remove_selected_confirm, count, count) : res.getQuantityString(R.plurals.caches_remove_all_confirm,count,count);
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/CacheListActivity.java]: I would prefer

```java
final String message = getQuantityString(adapter.getCheckedCount() > 0 ? R.plurals.caches_remove_selected_confirm : R.plurals.caches_remove_all_confirm, count, count);
```

to minimize repetitions.

**Comment by samueltardieu** [comment]
That looks great (I added a minor change request to minimize repetitions). Could you please change this and squash the commits together so that we have only one commit to merge, which logically represents the change?

Please ask us if you don't find how to do it.

**Comment by madankumarbaskaran** [comment]
Hi samueltardie,
                         Can you please check now?.
Thanks
Madan

**Comment by samueltardieu** [comment]
There are still minor style issues (missing spaces around "=" and ":" and after commas), but I will fix this by hand before merging since this is your first contribution and I want it in.

Here it is: faf4ae0a6c349c2049b1f6ee6422cab0a7aa6b76

**Comment by samueltardieu** [comment]
@Lineflyer I've merged this into `release` and I'll build a -RC, because otherwise handling translations will be a nightmare.

**Comment by madankumarbaskaran** [comment]
Thanks samueltardieu for your guidance.Will make sure not to repeat this issues.

**Pull request closed without merging**